### PR TITLE
Add job for BSK tests

### PR DESCRIPTION
**Pull request by SlayterDev**
Add a Job for PRs to run BSK tests.

Related to: https://app.asana.com/0/0/1204641354855486/f

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -31,3 +31,64 @@ jobs:
       - name: Test ddg2dnr
         working-directory: ./extension/packages/ddg2dnr
         run: npm test -- --grep Reference
+  BrowserServicesKit:
+    name: Run unit tests
```
Comment (GuiltyDolphin) [inline .github/workflows/test.yml:35]:
```suggestion
    name: BSK unit tests
```

**Comment by GuiltyDolphin** [comment]
@SlayterDev  I see a failure `ContentBlockerReferenceTests.swift:52 - failed - Rules were not compiled properly`, but not much more information. Could we make this more verbose to see the specific error message?

File: tracker-radar-tests/TR-domain-matching/domain_matching_tests.json [hunk 2]
```diff
@@ -400,7 +400,10 @@
                 "siteURL": "https://example2a.com",
                 "requestURL": "https://standard1.test/request-1a",
                 "requestType": "stylesheet",
-                "expectAction": "ignore"
+                "expectAction": "ignore",
+                "exceptPlatforms": [
+                    "ios-browser"
```
Comment (kdzwinel) [inline tracker-radar-tests/TR-domain-matching/domain_matching_tests.json:405]: Can we please document those exceptions in https://github.com/duckduckgo/privacy-reference-tests/tree/main/tracker-radar-tests/TR-domain-matching#platform-exceptions-1 ?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -31,3 +31,64 @@ jobs:
       - name: Test ddg2dnr
         working-directory: ./extension/packages/ddg2dnr
         run: npm test -- --grep Reference
+  BrowserServicesKit:
```
Comment (kdzwinel) [inline .github/workflows/test.yml:34]: I don't know github action syntax well enough to review, but @sammacbeth  did take a look and gave it a 👍 :) This is mostly copied from BSK repo, right?

Comment (SlayterDev) [reply .github/workflows/test.yml:34]: For the most part yes 👍

**Review by kdzwinel — APPROVED** [review]
LGTM, thank you both!

**Pull request merged**